wetterdienst/util/pandas.py

```diff
@@ -161,39 +207,68 @@ def export(self, target: str):
 
             Run database::
 
-                docker run --publish "8086:8086" influxdb:1.8.3
+                docker run -it --rm --publish=8086:8086 influxdb:1.8
 
             Acquire data::
 
-                wetterdienst dwd readings --station=1048,4411 --parameter=kl --resolution=daily --period=recent --target="influxdb://localhost/?database=dwd&table=weather"
+                wetterdienst dwd observations values --station=1048,4411 --parameter=kl --resolution=daily --period=recent --target="influxdb://localhost/?database=dwd&table=weather"
 
             Example queries::
 
                 http 'localhost:8086/query?db=dwd&q=SELECT * FROM weather;'
                 http 'localhost:8086/query?db=dwd&q=SELECT COUNT(*) FROM weather;'
             """
-            log.info(f"Writing to InfluxDB {database, tablename}")
+            log.info(f"Writing to InfluxDB. database={database}, table={tablename}")
             from influxdb.dataframe_client import DataFrameClient
 
-            # Setup the connection.
-            c = DataFrameClient(database=database)
-            c.create_database(database)
-
-            # Mungle the data frame.
+            # 1. Mungle the data frame.
+            # Use the "date" column as appropriate timestamp index.
             df = self.df.set_index(pd.DatetimeIndex(self.df["date"]))
             df = df.drop(["date"], axis=1)
-            df = df.dropna()
+
+            # Work around `ValueError: fill value must be in categories`.
+            # The reason is that the InfluxDB Pandas adapter tries to apply
+            # `tag_df.fillna('')  # replace NA with empty string`.
+            # However, it is not possible to apply `.fillna` to categorical
+            # columns. See:
+            # - https://github.com/pandas-dev/pandas/issues/24079
+            # - https://stackoverflow.com/questions/65316023/filling-np-nan-entries-of-float-column-gives-valueerror-fill-value-must-be-in-c/65316190
+            # - https://stackoverflow.com/questions/53664948/pandas-fillna-throws-valueerror-fill-value-must-be-in-categories
+            # - https://stackoverflow.com/questions/32718639/pandas-filling-nans-in-categorical-data/44633307
+            #
+            # So, let's convert all categorical columns back to their designated type representations.
+            # https://stackoverflow.com/questions/32011359/convert-categorical-data-in-pandas-dataframe/32011969#32011969
+            if "quality" in df:
+                df.quality = df.quality.astype("Int64")
+            categorical_columns = df.select_dtypes(["category"]).columns
+            df[categorical_columns] = df[categorical_columns].astype("str")
+
+            # When using the tidy format, don't export empty records.
+            if df.attrs.get("tidy"):
+                df = df.dropna()
 
             # Compute designated tag fields.
+            tag_columns = []
+            tag_candidates = [
```

Comment:
use Columns enum here!

Reply:
Thank you. I amended this place accordingly.